### PR TITLE
minPosition checks shifted to minIsolatedCollateral

### PR DESCRIPTION
**Pull request by 0xSpraggins**
# Description

Currently within Blueberry all leveraged positions have to pass a verification check that mandates the position value be within a certain bounds. This value is set within the `addStrategy` function. These checks are perfectly acceptable, but operationally we would prefer to check the isolated collateral value for setting a minimum requirement to enter a position instead of the overall position value. 

Changes Made:
- `minPositionSize` -> `minIsolatedCollateral` 
- Checks within the `BasicSpell` switched from validating that a position is above the minimum position value to validating that the isolated collateral's value is above the strategies `minIsolatedCollateral`.
- Error message `EXCEED_MIN_POS_SIZE` -> `BELOW_MIN_ISOLATED_COLLATERAL`
- Tests adapted to this change.

## Type of change

- [ ] Bug fix 
- [ ] New feature 
- [x] Breaking change 
- [ ] Dependency changes
- [ ] Code refactor / cleanup
- [ ] Documentation or wording changes
- [ ] Other

## Checklist:

- [x] The diff is legible and has no extraneous changes
- [x] Complex code has been commented, including external interfaces
- [x] Tests are included for all code paths
- [x] The base branch is either `main`, or there's a description of how to merge

### CODE REVIEW

**Review by Gornutz — APPROVED** [review]
- [X] @Gornutz Review
- [x] @mayorcoded Review
- [ ] @cuthalion0x Review

**Review by mayorcoded — APPROVED** [review]

File: contracts/interfaces/spell/IAuraSpell.sol [hunk 1]
```diff
@@ -13,10 +13,10 @@ interface IAuraSpell is IBasicSpell {
     /**
      * @notice Allows the owner to add a new strategy.
      * @param bpt Address of the Balancer Pool Token.
-     * @param minPosSize, USD price of minimum position size for given strategy, based 1e18
+     * @param minCollSize, USD price of minimum position size for given strategy, based 1e18
```
Comment (cuthalion0x) [inline contracts/interfaces/spell/IAuraSpell.sol:16]:
```suggestion
     * @param minCollSize, USD price of minimum isolated collateral for given strategy, based 1e18
```

File: contracts/interfaces/spell/IConvexSpell.sol [hunk 2]
```diff
@@ -33,10 +33,10 @@ interface IConvexSpell is IBasicSpell {
     /**
      * @notice Adds a new strategy to the spell.
      * @param crvLp Address of the Curve LP token for the strategy.
-     * @param minPosSize Minimum position size in USD for the strategy (with 1e18 precision).
+     * @param minCollSize Minimum position size in USD for the strategy (with 1e18 precision).
```
Comment (cuthalion0x) [inline contracts/interfaces/spell/IConvexSpell.sol:36]:
```suggestion
     * @param minCollSize Minimum isolated collateral in USD for the strategy (with 1e18 precision).
```

File: contracts/interfaces/spell/IIchiSpell.sol [hunk 3]
```diff
@@ -14,10 +14,10 @@ interface IIchiSpell is IBasicSpell {
     /**
      * @notice Adds a strategy to the contract.
      * @param vault Address of the vault linked to the strategy.
-     * @param minPosSize Minimum position size in USD, normalized to 1e18.
+     * @param minCollSize Minimum position size in USD, normalized to 1e18.
```
Comment (cuthalion0x) [inline contracts/interfaces/spell/IIchiSpell.sol:17]:
```suggestion
     * @param minCollSize Minimum isolated collateral in USD, normalized to 1e18.
```

File: contracts/interfaces/spell/IShortLongSpell.sol [hunk 4]
```diff
@@ -12,10 +12,10 @@ interface IShortLongSpell is IBasicSpell {
     /**
      * @notice Add strategy to the spell
      * @param swapToken Address of token for given strategy
-     * @param minPosSize USD price of minimum position size for given strategy, based 1e18
+     * @param minCollSize USD price of minimum position size for given strategy, based 1e18
```
Comment (cuthalion0x) [inline contracts/interfaces/spell/IShortLongSpell.sol:15]:
```suggestion
     * @param minCollSize USD price of minimum isolated collateral for given strategy, based 1e18
```

File: contracts/spell/BasicSpell.sol [hunk 5]
```diff
@@ -140,32 +140,32 @@ abstract contract BasicSpell is IBasicSpell, ERC1155NaiveReceiver, Ownable2StepU
      * @dev Internal function that appends to the strategies array.
      * @dev Emit {StrategyAdded} event.
      * @param vault The address of the vault associated with this strategy.
-     * @param minPosSize The minimum position size (USD value) for this strategy. Value is based on 1e18.
+     * @param minCollSize The minimum isolated Collateral size (USD value) for this strategy. Value is based on 1e18.
```
Comment (cuthalion0x) [inline contracts/spell/BasicSpell.sol:143]:
```suggestion
     * @param minCollSize The minimum isolated collateral size (USD value) for this strategy. Value is based on 1e18.
```

File: contracts/spell/BasicSpell.sol [hunk 6]
```diff
@@ -305,8 +305,13 @@ abstract contract BasicSpell is IBasicSpell, ERC1155NaiveReceiver, Ownable2StepU
         if (prevPosSize + addedPosSize > strategy.maxPositionSize) {
             revert Errors.EXCEED_MAX_POS_SIZE(strategyId);
         }
-        if (prevPosSize + addedPosSize < strategy.minPositionSize) {
-            revert Errors.EXCEED_MIN_POS_SIZE(strategyId);
+
+        // Get isolated collateral value
+        uint256 isolatedCollateralValue = bank.getIsolatedCollateralValue(bank.POSITION_ID());
+
+        // Check if isolated Collateral size is within bounds
```
Comment (cuthalion0x) [inline contracts/spell/BasicSpell.sol:312]:
```suggestion
        // Check if isolated collateral size is within bounds
```

**Review by cuthalion0x — APPROVED** [review]
Just code comment nits.

**Pull request merged**